sdk/appconfiguration/app-configuration/rollup.base.config.js

```diff
@@ -75,3 +93,77 @@ export function nodeConfig(test = false) {
 
   return baseConfig;
 }
+
+export function browserConfig(test = false) {
+  const baseConfig = {
+    input: input,
+    output: {
+      file: "dist-browser/appconfiguration.js",
+      format: "umd",
+      name: "Azure.AppConfiguration",
+      globals: {
+        "@azure/core-http": "Azure.Core.HTTP",
+        nock: "nock",
+        fs: "fs-extra"
+      },
+      sourcemap: true
+    },
+    external: ["nock", "fs-extra"],
+    preserveSymlinks: false,
+    plugins: [
+      sourcemaps(),
+      replace({
+        delimiters: ["", ""]
+      }),
+
+      shim({
+        constants: `export default {}`,
+        fs: `export default {}`,
+        os: `export default {}`,
+        dotenv: `export function config() { }`,
+        path: `export function join() {}`,
+        timers: `export default {}`,
+        stream: `export default {}`,
+        process: `export default {}`
```

Comment:
I think almost all of this is because of the recorder, actually. I can see if I can remove some of them - I built it up iteratively so some of them might be cruft now.

Reply:
Wow, you're going to love what I ended up with. I _only_ needed to shim dotenv after all was said and done.